CRM/Report/Form.php

```diff
@@ -1574,12 +1580,35 @@ public function addOptions() {
    */
   public function addChartOptions() {
     if (!empty($this->_charts)) {
-      $this->addElement('select', "charts", ts('Chart'), $this->_charts);
+      $this->addElement('select', "charts", ts('Chart'), $this->getChartLabels($this->_charts));
       $this->assign('charts', $this->_charts);
       $this->addElement('xbutton', $this->_chartButtonName, ts('View'), ['type' => 'submit']);
     }
   }
 
+  /**
+   * Map supported chart types to translated labels for use in the chart type select field
+   *
+   * @return array
+   */
+  private function getChartLabels() {
+    $charts = [];
+    $charts[''] = 'Tabular';
```

Comment:
ts() for Tabular?

It might be more "standard" to do the assignment of `_charts` in each subclass's constructor, and then you don't need this function, but this seems ok too.

Reply:
Oh yeah, extra `ts` added. I was so focused on then other two that I missed this one.

This approach seemed like it had the advantage that the strings couldn't get out of sync between each report - I have seen a few instances where repeated strings don't match in terms of spelling, capitalisation etc. But I'm not overly fussed either way.

Reply:
Hm, I have noticed that at least one third-party extension has instroduced their own keys, and other extensions are using the structure that CiviCRM core is using at the moment:

https://github.com/SyedShahiq/mainpeople/blob/master/sites/default/civicrm/extensions/net.ourpowerbase.report.advancedfundraising/CRM/Advancedfundraising/Form/Report/Contribute/Recovery.php

https://github.com/eileenmcnaughton/nz.co.fuzion.extendedreport/blob/master/CRM/Extendedreport/Form/Report/Contribute/Overview.php

I could extend `getChartLabels` to continue to support the old array strucutre, but might be that moving the labels to the constuctor of each report is cleaner whilst still retaining backwards compatiability. I'll have a think about the best way forward.

Reply:
Based on this thread I've reworked the code so that the translations are defined in the constructor of each report.